sansio_lsp_client/client.py

```diff
@@ -150,12 +259,57 @@ def _handle_response(self, response: Response) -> Event:
                 edits=parse_obj_as(t.List[TextEdit], response.result)
             )
 
+        elif request.method == "textDocument/hover":
+            event = Hover.parse_obj(response.result)
+            event.message_id = response.id
+
+        elif request.method == "textDocument/signatureHelp":
+            event = SignatureHelp.parse_obj(response.result)
+            event.message_id = response.id
+
+        elif request.method == "textDocument/documentSymbol":
+            result_type = t.get_type_hints(MDocumentSymbols)['result']
+            event = MDocumentSymbols(result=parse_obj_as(result_type, response.result))
```

Comment:
Is `get_type_hints` really necessary here? It seems a bit weird to me, and most of this code doesn't use it.